### PR TITLE
[WIP]support dynamic manager

### PR DESCRIPTION
**Pull request by leizhiyuan**
### Motivation:

Explain the context, and why you're making that change.
To make others understand what is the problem you're trying to solve.

### Modification:

Describe the idea and modifications you've done.

### Result:

Fixes #<GitHub issue number>. 

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

File: core/api/src/main/java/com/alipay/sofa/rpc/dynamic/DynamicManager.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     */
+    public abstract Properties initServiceConfigutration(String service);
+
+    /**
+     * fetch the value of the specify key
+     * @param service
+     * @param key
+     * @return
+     */
+    public abstract String fetchKey(String service, String key);
```
Comment (ujjboy) [inline core/api/src/main/java/com/alipay/sofa/rpc/dynamic/DynamicManager.java]: getValue() is better.

File: core/api/src/main/java/com/alipay/sofa/rpc/dynamic/DynamicManagerFactory.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    /**
+     * 得到动态配置管理
+     *
+     * @param alias 别名
+     * @return DynamicManager 实现
+     */
+    public static synchronized DynamicManager getDynamicManager(String appName, String alias) {
+        if (ALL_DYNAMICS.size() > 3) { // 超过3次 是不是配错了？
+            if (LOGGER.isWarnEnabled()) {
+                LOGGER.warn("Size of registry is greater than 3, Please check it!");
```
Comment (ujjboy) [inline core/api/src/main/java/com/alipay/sofa/rpc/dynamic/DynamicManagerFactory.java]: not registry.

File: core/api/src/main/java/com/alipay/sofa/rpc/dynamic/DynamicManager.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    protected DynamicManager(String appName) {
+        this.appName = appName;
+    }
+
+    /**
+     * init service ,which service is servicename:1.0
+     * @param service
+     * @return
+     */
+    public abstract Properties initServiceConfigutration(String service);
```
Comment (ujjboy) [inline core/api/src/main/java/com/alipay/sofa/rpc/dynamic/DynamicManager.java]: why return `Properties` ?

**Comment by leizhiyuan** [comment]
please do not review it now. we only test it for a demo

**Comment by ujjboy** [comment]
There is too much incompatible API in this branch.  Change millstone to 5.6.0.

**Pull request closed without merging**